src/stdlib/array.hli

```diff
@@ -17,6 +17,12 @@ module Cursor : sig
                               and type ''a $t := ''a $t
 end
 
+val hash_fold: (''a -> Hash.State.t -> Hash.State.t) -> ''a $t -> Hash.State.t
+    -> Hash.State.t
+(** [hash_fold hash_fold_a t state] incorporates the hash of [t] into [state]
+    and returns the resulting state.  Array elements are sequentially
+    hash-folded into the resulting state via [hash_fold_a]. *)
+
```

Comment:
It might be a good idea to extract this into a polymorphic functor in the long run, but it's not yet clear to me what level to integrate that at.  This will be something to revisit when adding hashing for `'a list`.